enzy_htp/structure/residue.py

```diff
@@ -301,6 +306,12 @@ def is_connected(self) -> bool:
     def has_init_charge(self) -> bool:
         """check if self has charge"""
         return math.prod([atom.has_init_charge() for atom in self.atoms])  
+    
+    def has_hydrogen(self) -> bool:
```

Comment:
The grammar doesn't quite make sense. instead of `has_hydrogen()`, try `has_hydrogens()`. Saying "self" in the docstring is also not very helpful. A good rule of thumb with functions that return `bool`'s is to write docstrings as questions. Consider a rewrite similar to this:
```
def has_hydrogens(self) -> bool:
    """Does the residue contain hydrogen atoms?"""
    pass
```

Reply:
Rewrited!